### PR TITLE
Change the namespacing model for Aerospike

### PR DESCRIPTION
**Pull request by geobeau**
Previously we were starting one probe worker for each cluster. With one driver per worker.

Now we spawn one worker per namespace per cluster.
The main benefit is that namespace are handled
independantly by the probe, it won't get stuck if
one namespace is failing to start.

### CODE REVIEW

File: pkg/aerospike/checks.go [hunk 1]
```diff
@@ -186,50 +184,48 @@ func DurabilityPrepare(p topology.ProbeableEndpoint) error {
...
+
+	recVal, err := e.Client.Get(&policy.BasePolicy, allPushedFlag)
 
-		recVal, err := e.Client.Get(&policy.BasePolicy, allPushedFlag)
+	if err != nil && !err.Matches(as.ErrKeyNotFound.ResultCode) {
+		return err
+	}
+	// If the flag was found we skip the init as it has already been done
+	if recVal != nil && recVal.Bins["val"] == expectedAllPushedFlagVal {
+		return nil
```
Comment (pgoron) [inline pkg/aerospike/checks.go:200]: Is this change expected ? it doesn't seem related to the intent of the commit.

**Review by pgoron — COMMENTED** [review]
I'm overall ok with your change but a bit concerned that no UT impacted. Do we have a good code coverage ?

File: pkg/aerospike/checks.go (hunk 1 above)
Comment (geobeau) [reply pkg/aerospike/checks.go:200]: it reapplies what was previously on L202

File: probes/aerospike/README.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Checks
 
-All the checks are executed for each Aerospikes namespaces has they are isolated.
+The probe is considering the namespaces as different "clusters". This is done
+because namespaces are mostly isolated and can have different configuration.
```
[the remark this answers is not in the document]
Comment (geobeau) [reply probes/aerospike/README.md]: is it better now?

**Review by pgoron — APPROVED** [review]

File: probes/aerospike/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pgoron) [reply probes/aerospike/README.md]: yes

**Comment by geobeau** [comment]
For UT, most of them are on the framework which wasn't touched by the review.
Aerospike probe doesn't have UT, the best way to manage testing would be integration testing but I'm not sure we can easily start an aerospike server.
The framework itself would benefit more UT as well.
